base/cpuid.jl

```diff
@@ -88,14 +89,29 @@ function normalize_arch(arch::String)
     return arch
 end
 
+const ALL_FEATURES = let
+    get_features(prefix::String) =
+        getfield.(Ref(@__MODULE__), filter(n -> startswith(String(n), prefix), (names(@__MODULE__; all=true))))
+    Dict(
+        "i686" => get_features("JL_X86"),
+        "x86_64" => get_features("JL_X86"),
+        "armv6l" => get_features("JL_AArch32"),
+        "armv7l" => get_features("JL_AArch32"),
+        "aarch64" => get_features("JL_AArch64"),
+        "powerpc64le" => UInt32[],
+    )
+end
+
+# Use `@eval` to statically determine the list of features for the current architecture.
+@eval function cpu_isa()
+    return ISA(Set{UInt32}(feat for feat in $(ALL_FEATURES[normalize_arch(String(Sys.ARCH))]) if test_cpu_feature(feat)))
+end
```

Comment:
I realised we can avoid always recomputing the list of features for the current architecture and just inline it at precompile time.  On my laptop, before:
```
julia> @benchmark CPUID.cpu_isa()
BenchmarkTools.Trial: 10000 samples with 48 evaluations.
 Range (min … max):  895.604 ns … 90.301 μs  ┊ GC (min … max): 0.00% … 98.34%
 Time  (median):     984.552 ns              ┊ GC (median):    0.00%
 Time  (mean ± σ):     1.153 μs ±  2.866 μs  ┊ GC (mean ± σ):  9.56% ±  3.80%

   ▃▆█▆▃▁                                                       
  ▇██████▇▆▄▃▂▂▂▂▂▂▂▂▂▂▂▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁ ▂
  896 ns          Histogram: frequency by time            2 μs <

 Memory estimate: 1.41 KiB, allocs estimate: 17.
```
after:
```
julia> @benchmark CPUID.cpu_isa()
BenchmarkTools.Trial: 10000 samples with 155 evaluations.
 Range (min … max):  679.871 ns …  18.916 μs  ┊ GC (min … max): 0.00% … 89.46%
 Time  (median):     745.200 ns               ┊ GC (median):    0.00%
 Time  (mean ± σ):   849.687 ns ± 709.196 ns  ┊ GC (mean ± σ):  4.12% ±  4.95%

  ▁▆█▇▅▃▃▃▃▃▃▂▂▁▁▁                                           ▂▃ ▂
  █████████████████▇█▇▇▇▆▆▇▆▆▆▆▆▆▄▅▅▄▁▅▄▅▅▅▃▁▁▄▁▃▁▁▃▄▃▃▄▃▁▁▄▆██ █
  680 ns        Histogram: log(frequency) by time       1.97 μs <

 Memory estimate: 848 bytes, allocs estimate: 7.
```

Reply:
Nice!  Less allocations, always a good thing!

Reply:
With the latest version:
```
julia> @benchmark Base.BinaryPlatforms.CPUID.cpu_isa()
BenchmarkTools.Trial: 10000 samples with 196 evaluations.
 Range (min … max):  480.342 ns …  12.980 μs  ┊ GC (min … max): 0.00% … 94.89%
 Time  (median):     527.505 ns               ┊ GC (median):    0.00%
 Time  (mean ± σ):   598.004 ns ± 670.168 ns  ┊ GC (mean ± σ):  6.65% ±  5.69%

   ▄▆███▇▆▄▄▄▃▄▃▂▂▂▂▁▁▁▁▁▁▁▂▁▁▁▂▁                               ▂
  ▆██████████████████████████████▇▇▇▆█▇█▆▇▆▇▇▇▆▅▅▇▇▅▅▃▆▆▃▆▆▅▄▄▅ █
  480 ns        Histogram: log(frequency) by time          1 μs <

 Memory estimate: 848 bytes, allocs estimate: 7.
```
I believe it's a bit faster because the new version collects only the features we are interested in, instead of _all_ of those for the given architecture, so we're just doing fewer iterations.  The new version is also closer in spirit to what we're currently doing.